data-prepper-plugins/mongodb/src/test/java/org/opensearch/dataprepper/plugins/mongo/documentdb/MongoTasksRefresherTest.java

```diff
@@ -0,0 +1,231 @@
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+import static org.opensearch.dataprepper.plugins.mongo.documentdb.MongoTasksRefresher.CREDENTIALS_CHANGED;
+import static org.opensearch.dataprepper.plugins.mongo.documentdb.MongoTasksRefresher.EXECUTOR_REFRESH_ERRORS;
+
+@ExtendWith(MockitoExtension.class)
+class MongoTasksRefresherTest {
+    private static final String TEST_USERNAME = "test_user";
+    private static final String TEST_PASSWORD = "test_password";
+
+    @Mock
+    private EnhancedSourceCoordinator enhancedSourceCoordinator;
+
+    @Mock
+    private Buffer<Record<Event>> buffer;
+
+    @Mock
+    private Function<Integer, ExecutorService> executorServiceFunction;
+
+    @Mock
+    private AcknowledgementSetManager acknowledgementSetManager;
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private MongoDBSourceConfig sourceConfig;
+
+    @Mock
+    private MongoDBSourceConfig.AuthenticationConfig credentialsConfig;
+
+    @Mock
+    private ExecutorService executorService;
+
+    @Mock
+    private CollectionConfig collectionConfig;
+
+    @Mock
+    private Counter credentialsChangeCounter;
+
+    @Mock
+    private Counter executorRefreshErrorsCounter;
+
+    private MongoTasksRefresher createObjectUnderTest() {
+        return new MongoTasksRefresher(
+                buffer, enhancedSourceCoordinator, pluginMetrics, acknowledgementSetManager, executorServiceFunction);
+    }
+
+    @BeforeEach
+    void setUp() {
+        when(executorServiceFunction.apply(anyInt())).thenReturn(executorService);
+        when(sourceConfig.getCollections()).thenReturn(List.of(collectionConfig));
+        when(collectionConfig.isExport()).thenReturn(true);
+        when(collectionConfig.isStream()).thenReturn(true);
+    }
+
+    @Test
+    void testUpdateWithBasicAuthUnchanged() {
+        final MongoTasksRefresher objectUnderTest = createObjectUnderTest();
+        objectUnderTest.initialize(sourceConfig);
+        verify(executorServiceFunction).apply(eq(3));
+        when(sourceConfig.getCredentialsConfig()).thenReturn(credentialsConfig);
+        when(credentialsConfig.getUsername()).thenReturn(TEST_USERNAME);
+        when(credentialsConfig.getPassword()).thenReturn(TEST_PASSWORD);
+        final MongoDBSourceConfig newSourceConfig = mock(MongoDBSourceConfig.class);
+        final MongoDBSourceConfig.AuthenticationConfig newCredentialsConfig = mock(
+                MongoDBSourceConfig.AuthenticationConfig.class);
+        when(newSourceConfig.getCredentialsConfig()).thenReturn(newCredentialsConfig);
+        when(newCredentialsConfig.getUsername()).thenReturn(TEST_USERNAME);
+        when(newCredentialsConfig.getPassword()).thenReturn(TEST_PASSWORD);
+        objectUnderTest.update(newSourceConfig);
+        verify(executorService, times(3)).submit(any(Runnable.class));
```

Comment:
Can you check the specific runnable classes here, i.e. ExportScheduler, ExportWorker, StreamScheduler?

Similarly for other test cases.